File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;
class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        PriorityQueue<Integer> top = new PriorityQueue<>();
        int index = 0;
        while(top.size() < k){
            if(index == score.length){
                break;
            }
            top.offer(score[index]);
            answer[index] = top.peek();
            index++;
        }
        while(index < score.length){
            if(score[index] > top.peek()){
                top.poll();
                top.offer(score[index]);
            }
            answer[index] = top.peek();
            index++;
        }
        return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    top = []
    index = 0
    
    while len(top) < k:
        if index == len(score):
            break
        top.append(score[index])
        top.sort()
        answer.append(top[0])
        index += 1
    
    while index < len(score):
        if score[index] > top[0]:
            top.remove(top[0])
            top.append(score[index])
            top.sort()
        answer.append(top[0])
        index += 1
    
    return answer